query-engine/connector-test-kit-rs/query-engine-tests/tests/new/occ.rs

```diff
@@ -0,0 +1,389 @@
+        }
+    }
+
+    async fn book_unclaimed_seat(user_id: u64, seat_id: u64, runner: Arc<Runner>) -> (u64, u64) {
+        let query = indoc::formatdoc!(
+            r##"
+                      mutation {{
+                          updateManySeat(
+                              data: {{ userId: {user_id}, version: {{ increment: 1 }} }},
+                              where: {{ id: {seat_id}, version: 0 }}
+                          )
+                          {{ count }}
+                      }}
+                    "##
+        );
+        let response = runner.query(query).await.unwrap().to_json_value();
+        let seat_count = response["data"]["updateManySeat"]["count"].as_u64().unwrap();
+        (user_id, seat_count)
+    }
+
+    async fn book_seat_for_user(user_id: u64, runner: Arc<Runner>) -> (u64, u64) {
+        let (seat_id, _version) = find_unclaimed_seat(runner.clone()).await;
+        book_unclaimed_seat(user_id, seat_id, runner).await
+    }
+
+    async fn delete_seats(runner: Arc<Runner>) {
+        let delete_seats = r#"
+            mutation {
+                deleteManySeat(where: {}) {
+                count
+                }
+            }
+        "#;
+        runner.query(delete_seats).await.unwrap().assert_success();
+    }
+
+    async fn delete_users(runner: Arc<Runner>) {
+        let delete_users = r#"
+            mutation {
+                deleteManyUser(where: {}) {
+                count
+                }
+            }
+        "#;
+        runner.query(delete_users).await.unwrap().assert_success();
+    }
+
+    async fn run_occ_reproduce_test(runner: Arc<Runner>) {
+        const USERS_COUNT: u64 = 5;
+
+        create_one_seat(runner.clone()).await;
+
+        for i in 0..=USERS_COUNT {
+            create_one_user(i, runner.clone()).await;
+        }
+
+        let mut set = tokio::task::JoinSet::new();
+        for user_id in 0..=USERS_COUNT {
+            set.spawn(book_seat_for_user(user_id, runner.clone()));
+        }
+
+        let mut booked_user_id = 100;
+        let mut total_booked = 0;
+        while let Some(res) = set.join_next().await {
+            let (user_id, count) = res.unwrap();
+
+            if count > 0 {
+                total_booked += count;
+                booked_user_id = user_id;
+            }
+        }
+
+        assert_eq!(total_booked, 1);
+
+        let booked_seat = runner
+            .query("query { findFirstSeat { id version userId } }")
+            .await
+            .unwrap()
+            .to_json_value();
+
+        let found_booked_user_id = booked_seat
+            .as_object()
+            .unwrap()
+            .get("data")
+            .unwrap()
+            .as_object()
+            .unwrap()
+            .get("findFirstSeat")
+            .unwrap()
+            .as_object()
+            .unwrap()
+            .get("userId")
+            .unwrap()
+            .as_u64()
+            .unwrap();
```

Comment:
Same comment as above here. Either use `run_query_json!` or use the index accessor. Are you not using any macros because you want to reduce their usage?